tensorflow/python/ops/special_math_ops.py

```diff
@@ -324,13 +324,43 @@ def sort_key(input_index, a):
     num_summed_elements = _total_size(t0_shape[-len(axes_to_sum):])
     new_shape = t0_shape[:len(preserved_axes)] + (num_broadcast_elements_t0,
                                                   num_summed_elements)
+    # Make new_shape valid, if invalid to be used in reshape
+    none_dims = [i for i, d in enumerate(new_shape)
```

Comment:
TF implements constant folding, which means that if a shape() node that exists in the graph can be computed statically from constants, it will be done once during the first run and never again for future computations, so concerns about performance here may not matter.